packages/block-editor/src/components/colors-gradients/style.scss

```diff
@@ -77,18 +77,33 @@ $swatch-gap: 12px;
 	border-right: 1px solid $gray-300;
 	border-bottom: 1px solid $gray-300;
 
-	&.first {
+	// Identify the first instance as the item that is not a subsequent item to an
+	// item with the same class. The `~` selector is used instead of `+` to take
+	// into account any potential `ToolsPanelItem` placeholder item.
+	&:not(& ~ &) {
 		margin-top: $grid-unit-30;
 		border-top-left-radius: $radius-block-ui;
 		border-top-right-radius: $radius-block-ui;
 		border-top: 1px solid $gray-300;
 	}
 
-	&.last {
+	// Identify the last instance as an item without any subsequent siblings
+	// with the same class. The `~` selector is used instead of `+` to take
+	// into account any potential `ToolsPanelItem` placeholder item.
+	&:not(:has(~ &)) {
 		border-bottom-left-radius: $radius-block-ui;
 		border-bottom-right-radius: $radius-block-ui;
 	}
 
+	// Less accurate fallback to detect the last instance of this item
+	// for older browsers not yet supporting the `:has()` selector.
+	@supports not selector(:has(*)) {
+		&.last {
+			border-bottom-left-radius: $radius-block-ui;
+			border-bottom-right-radius: $radius-block-ui;
+		}
+	}
+
```

Comment:
```suggestion
```

If we remove the use of `:has` in the selectors above we should also be able to get rid of this fallback CSS.